lib/src/analysis_options.dart

```diff
@@ -25,74 +32,85 @@ class AnalysisOptions {
       @required this.rules});
 
   factory AnalysisOptions.from(String content) {
+    try {
+      final node = loadYamlNode(content ?? '');
+
+      return AnalysisOptions.fromMap(
+          node is YamlMap ? yamlMapToDartMap(node) : {});
+    } on YamlException catch (e) {
+      throw FormatException(e.message, e.span);
+    }
+  }
+
+  factory AnalysisOptions.fromMap(Map<String, Object> configMap) {
     Config metricsConfig;
-    var metricsExcludePatterns = <String>[];
+    Iterable<String> metricsExcludePatterns = <String>[];
     var rules = <String, Map<String, Object>>{};
 
-    final node = loadYamlNode(content ?? '');
-    if (node is YamlMap && node.nodes[_rootKey] is YamlMap) {
-      final metricsOptions = node.nodes[_rootKey] as YamlMap;
-
-      if (_isYamlMapOfStringsAndIntegers(metricsOptions.nodes[_metricsKey])) {
-        final configMap = metricsOptions[_metricsKey] as YamlMap;
+    final metricsOptions = configMap[_rootKey];
+    if (metricsOptions != null && metricsOptions is Map<String, Object>) {
+      final configMap = metricsOptions[_metricsKey];
+      if (configMap != null && configMap is Map<String, Object>) {
```

Comment:
May be move this check to separate function like `isMap` for all same usages on lines 51, 53, 83, 100?

Reply:
fixed